src/games/kalambury/KalamburyBoard.js

```diff
@@ -89,62 +128,187 @@ const KalamburyBoard = ({ G, ctx, playerID, moves, gameMetadata }) => {
       <Container style={styles.mainContent}>
         <Grid>
           <Grid.Column width="12">
-            {isDrawing ? (
-              <DrawingBoard playerData={playerData} {...{ G, ctx, moves }} />
-            ) : (
-              <GuessingBoard
-                envokeLastAnswer={envokeLastAnswer}
-                previousUserGuesses={getUserGuesses(guesses, playerID)}
-                playerID={playerID}
-                guessInputRef={guessInputRef}
-                guess={guess}
-                setGuess={setGuess}
-                {...{ G, ctx, moves }}
+            {header()}
+            {hasGameStarted ? (
+              <GameBoard
+                {...{
+                  G,
+                  ctx,
+                  moves,
+                  playerData,
+                  isDrawing,
+                  envokeLastAnswer,
+                  getUserActions,
+                  actions,
+                  playerID,
+                  guessInputRef,
+                  guess,
+                  setGuess,
+                  remainingTime,
+                  canChangePhrase,
+                  ChangePhrase,
+                }}
               />
-            )}
-            <Header as="h3" textAlign="center" style={{ marginTop: 0 }}>
-              {remainingTime()}
-            </Header>
-            {isDrawing ? (
-              <Segment basic textAlign="center">
-                <Button color="yellow" disabled={!canChangePhrase} onClick={() => ChangePhrase()}>
-                  Get new phrase
-                </Button>
-              </Segment>
             ) : (
-              <></>
+              <WaitingBoard
+                previousUserMessages={getUserActions(actions, playerID, "message")}
+                {...{ moves, canManageGame, setGuess, guess }}
+              />
             )}
           </Grid.Column>
           <Grid.Column width="4" style={{ marginTop: "19px" }}>
             <KalamburySidebar
               handleGuessClick={handleGuessClick}
-              getUserGuesses={getUserGuesses}
+              getUserActions={getUserActions}
               {...{ G, ctx, playerID, moves, gameMetadata }}
             />
           </Grid.Column>
         </Grid>
       </Container>
-    </div>
+    </>
   );
 };
 
+const WaitingBoard = ({
```

Comment:
this file is way too big, extract `WaitingBoard`, and `GameBoard` (with it's `Drawing` and `Guessing` boards) to separate files